LXStudio-IDE/src/main/java/titanicsend/model/TEPanelModel.java

```diff
@@ -1,52 +1,85 @@
 package titanicsend.model;
 
 import java.util.ArrayList;
+
 import heronarts.lx.model.LXModel;
 import heronarts.lx.model.LXPoint;
+import titanicsend.app.TEVirtualColor;
 
 public class TEPanelModel extends LXModel {
-  public final TEVertex v1, v2, v3;
-  public final TEEdgeModel e1, e2, e3;
+  public final static String NONEXISTENT = "nonexistent";
+  public final static String SOLID = "solid";
+  public final static String LIT = "lit";
+
+  public final TEVertex v0, v1, v2;
+  public final TEEdgeModel e0, e1, e2;
+  public final LXPoint centroid;
+  public String panelType;
+
+  // Set to non-null and the virtual display will shade the panel's triangle
+  public TEVirtualColor virtualColor;
 
   // Given an Edge and three Vertexes, return the number of vertexes the edge touches
-  private int count_touches(TEEdgeModel e, TEVertex v1, TEVertex v2, TEVertex v3) {
+  private int count_touches(TEEdgeModel e, TEVertex v0, TEVertex v1, TEVertex v2) {
     int rv = 0;
+    if (e.touches(v0)) rv++;
     if (e.touches(v1)) rv++;
     if (e.touches(v2)) rv++;
-    if (e.touches(v3)) rv++;
     return rv;
   }
 
-  public TEPanelModel(ArrayList<LXPoint> points, TEVertex v1, TEVertex v2, TEVertex v3,
-                      TEEdgeModel e1, TEEdgeModel e2, TEEdgeModel e3) {
+  public TEPanelModel(ArrayList<LXPoint> points, TEVertex v0, TEVertex v1, TEVertex v2,
+                      TEEdgeModel e0, TEEdgeModel e1, TEEdgeModel e2, String panelType,
+                      LXPoint centroid) {
     super(points);
 
+    assert (panelType.equals(NONEXISTENT) ||
+            panelType.equals(SOLID) ||
+            panelType.equals(LIT));
+    this.panelType = panelType;
+
+    this.centroid = centroid;
+
+    if (panelType.equals(NONEXISTENT)) {
+      // Display nonexistent panels as wispy pink
+      this.virtualColor = new TEVirtualColor(255, 0, 255, 100);
```

Comment:
attaching this to a TEComponent boolean might be nice

Reply:
I'm hoping to get all the nonexistent panels out of the datafile soon.

Reply:
I have a small PR to add a toggle for now..